fix: allow dev admin editors to see the chart they just created

realSpok opened the pull request:
## Definition of Done

### General

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [x] Test manually the implemented changes
- [x] Validate the code quality (indentation, syntax, style, simplicity, readability)
- [x] Ensure that Types have been updated according to your changes (if needed)

### Security

- [x] Consider the security impact of the changes made

romain-gilliotte commented on packages/agent/src/agent/services/permissions.ts:
```diff
@@ -31,6 +31,13 @@ export default class PermissionService {
 
+    // If the permissions level already allow the chart, no need to check further
+    const CHARTS_ALLOWED_PERMISSIONS_LEVELS = ['admin', 'editor', 'developer'];
+
```
```
  /** Checks that a charting query is in the list of allowed queries */
  async canChart(context: Context): Promise<void> {
    const chart = { ...context.request.body };

    // If the permissions level already allow the chart, no need to check further
    if (!['admin', 'editor', 'developer'].includes(context.state.user.permissionLevel)) {
      // When the server sends the data of the allowed charts, the target column is not specified
      // for relations => allow them all.
      if (chart?.group_by_field?.includes(':'))
        chart.group_by_field = chart.group_by_field.substring(0, chart.group_by_field.indexOf(':'));

      const chartHash = hashObject(chart, {
        respectType: false,
        excludeKeys: key => chart[key] === null,
      });

      await this.can(context, `chart:${chartHash}`);
    }
  }
```

realSpok commented:
As discussed, I removed `const CHARTS_ALLOWED_PERMISSIONS_LEVELS`,  and moved the `const chart = { ...context.request.body };` to where it is needed.
merging now
